domain/schema/model.go

```diff
@@ -199,6 +203,29 @@ BEGIN
 END;`, tableAgentVersion))
 	})
 
+	// Add a custom namespace that only watches for insert and delete operations
+	// for units. This is used, for example, by the deployer which manages running
+	// agents, so only needs to know about new units, or removed unites.
+	patches = append(patches, func() schema.Patch {
+		return schema.MakePatch(fmt.Sprintf(`
+INSERT INTO change_log_namespace VALUES (%[1]d, 'unit_insert_delete', 'Unit insert or delete changes only');
+
+CREATE TRIGGER trg_log_unit_insert_delete_insert
+AFTER INSERT ON unit FOR EACH ROW
+BEGIN
+    INSERT INTO change_log (edit_type_id, namespace_id, changed, created_at)
+    VALUES (1, %[1]d, NEW.name, DATETIME('now'));
+END;
+
+CREATE TRIGGER trg_log_unit_insert_delete_delete
+AFTER DELETE ON unit FOR EACH ROW
+BEGIN
+    INSERT INTO change_log (edit_type_id, namespace_id, changed, created_at)
+    VALUES (4, %[1]d, OLD.name, DATETIME('now'));
+END;
```

Comment:
I have concerns about the delete logic. The unit should be cleaned up from the machine when it's dead. Not after it's been deleted from the database. 

This logic is now broken: https://github.com/juju/juju/blob/main/internal/worker/deployer/deployer.go#L167-L175

Reply:
Not quite. We have to cater for `juju remove-unit`.

Reply:
> This logic is now broken: https://github.com/juju/juju/blob/main/internal/worker/deployer/deployer.go#L167-L175

Not quite, this should still work, because of https://github.com/juju/juju/blob/73e1adb9d014ebb43c100ab922295d684de2e247/internal/worker/deployer/deployer.go#L160-L161